runtime/coverage.go

```diff
@@ -505,7 +517,8 @@ func (r *CoverageReport) MarshalLCOV() ([]byte, error) {
 	buf := new(bytes.Buffer)
 	for _, location := range locations {
 		coverage := r.Coverage[location]
-		_, err := fmt.Fprintf(buf, "TN:\nSF:%s\n", location.ID())
+		locationSource := r.locationSource(location)
```

Comment:
I think the `LCOV` format will consider any report problematic, if `SF` does not contain a legal filename.
I am doing some `sed` manipulation, in order to replace any `AddressLocation`, see: https://github.com/m-Peter/flow-code-coverage/blob/main/normalize_coverage_report.sh
Would it make sense to drop any location for which there is not location source file?

Reply:
When would a location be not a legal file name? AFAICS, the IDs of location kinds themselves are legal, e.g. `A.<hex-address>` for address locations, and the qualified identifier suffix is always a Cadence identifier (`[A-Za-z0-9_]+`), plus dots, which is also legal in filenames

Reply:
They might be legal, but they cannot be utilized by 3rd party tools that consume code coverage reports in LCOV format. For example, Codecov cannot translate an address location to its source file. And as such, it cannot provide visualizations. I think all of these tools need filepaths, in order to open them and display their source code.